src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1017,16 +1017,24 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
 
             for (ObjectList::const_iterator itr = targets->begin(); itr != targets->end(); ++itr)
             {
-                if (!IsCreature(*itr))
+                if (!IsCreature(*itr) || !IsGameObject(*itr))
```

Comment:
was this code tested ingame and debugged line by line with both a Creature, a GameObject and an object which is not any of these ?